feat: Implemented about and countdown page v2

raywa04 opened the pull request:
cleaned up original pr (270)

raywa04 committed:
feat: Implemented about and countdown page v2

uy-seng commented on components/homeComponents/HackCountdown.tsx:
the countdown ui is not showing fully in mobile view

uy-seng replied:
all css module in this file can be separated into  another css module

uy-seng commented on components/homeComponents/HackCountdown.tsx:
```diff
@@ -0,0 +1,209 @@
+};
+
+const Hack24Countdown: React.FC = () => {
+  return <HackCountdown targetDate="2024-11-04T00:00:00Z" />;
```
i think it would be better to set targetDate as part of `hackportal.config.ts` and then import it and use as constant for the props

uy-seng commented on components/homeComponents/HomeAbout.tsx:
same for this component, css module can be separated instead

uy-seng commented on components/homeComponents/HomeAbout.tsx:
```diff
@@ -1,25 +1,187 @@
+          }
+        `}
+      </style>
+      <h1 style={styles.header}>About HackPortal</h1>
```
this component overlap with `<p>` below it in mobile view

uy-seng commented on components/homeComponents/HomeAbout.tsx:
```diff
@@ -1,25 +1,187 @@
+        Hackathons are 24-hour gatherings where students collaborate to <br /> create innovative
+        projects, forge new connections, and compete for prizes.
+      </p>
+      <div className="statsContainer" style={styles.statsContainer}>
```
`div` inside container overlapping inside mobile view

raywa04 committed:
Update HackCountdown.tsx

fixed the issue with mobile view not being fully portrayed. will clean up code (separate the css module) next

raywa04 committed:
Update HomeAbout.tsx

like the countdown page, mobile view is fixed here. just need to clean up the code.

raywa04 committed:
Update HomeAbout.tsx

fixed issue with background image appearing over the designated web page

raywa04 committed:
Update HomeAbout.tsx

DedsecKnight commented on components/homeComponents/HackCountdown.tsx:
```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect, CSSProperties } from 'react';
+
+interface Hack24CountdownProps {
```
The typename should be changed to something else since this is generic hackportal.

DedsecKnight commented on components/homeComponents/HackCountdown.tsx:
```diff
@@ -0,0 +1,203 @@
+    fontFamily: 'Poppins',
+    fontWeight: 400,
+    fontSize: '4vw',
+    lineHeight: '5vw',
+    color: '#05149C',
+    marginTop: '1vw',
+  };
+
+  const styles: { [key: string]: CSSProperties } = {
+    countdownContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      flexDirection: 'column',
+      fontFamily: 'Arial, sans-serif',
+      color: '#2E0480',
+      background:
+        '#FFFFFF url("data:image/svg+xml,%3Csvg width%3D%2250vw%22 height%3D%22353px%22 viewBox%3D%220 0 100vw 706px%22 fill%3D%22none%22 xmlns%3D%22http%3A//www.w3.org/2000/svg%22%3E%3Cdefs%3E%3Cfilter id%3D%22blurStrong%22%3E%3CfeGaussianBlur stdDeviation%3D%2220%22/%3E%3C/filter%3E%3C/defs%3E%3Cellipse cx%3D%2286.5vw%22 cy%3D%22260px%22 rx%3D%22180px%22 ry%3D%22180px%22 fill%3D%22%234A3AFF%22  ...
+      height: '100vh',
+      padding: '2vw',
+      position: 'relative',
+      overflow: 'hidden',
+    },
+    timeSection: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      gap: '1.5vw',
+    },
+    notifySection: {
+      marginTop: '2vw',
+      textAlign: 'center',
+      fontFamily: 'Poppins',
+      fontWeight: 400,
+      fontSize: '2.5vw',
+      lineHeight: '3.5vw',
+      color: '#000000',
+      width: '90%',
+    },
+  };
```
nit: any reason why we're using pure CSS here instead of tailwindcss?

raywa04 replied:
I found it a lot easier to program specific details with pure CSS. I can implement it in tailwindcss if you want me to.

DedsecKnight replied:
it's just nit so I think refactor later if needed to.

DedsecKnight commented on components/homeComponents/HomeAbout.tsx:
```diff
@@ -1,25 +1,223 @@
+      fontFamily: 'Arial, sans-serif',
+      textAlign: 'center',
+      width: '100vw',
+      height: '130vh',
+      overflow: 'hidden',
+      paddingTop: '2vh',
+    },
+    header: {
+      fontFamily: 'Fredoka, sans-serif',
+      fontWeight: 600,
+      fontSize: 'calc(24px + 2vw)',
+      color: '#05149C',
+      padding: '1vh 0',
+    },
+    description: {
+      fontFamily: 'DM Sans, sans-serif',
+      fontWeight: 400,
+      fontSize: '16px',
+      color: '#000000',
+      width: '90%',
+      margin: '1vh 0',
+    },
+    statsContainer: {
+      position: 'relative',
+      width: '100%',
+      height: '100%',
+      fontFamily: 'DM Sans, sans-serif',
+      fontWeight: 400,
+      fontSize: '30px',
+      display: 'flex',
+      flexWrap: 'wrap',
+      justifyContent: 'space-around',
+      marginTop: '2vh',
+    },
+    statsItem: {
+      width: '45%',
+      margin: '2% 0',
+      position: 'relative',
+    },
+  };
```
Same question as above.

raywa04 committed:
fix: Cleaned up files, took into account comments as well

raywa04 committed:
fix: Resolved conflicts

raywa04 committed:
Merge branch 'develop' into CountdownAboutPages

DedsecKnight committed:
Merge branch 'develop' into CountdownAboutPages

DedsecKnight reviewed (approved):
lgtm :+1:. Will merge this in now since there are 2 approvals already.
